src/Apps/Legion/Components/StoreFormModal/StoreFormModal.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useMemo, useState } from 'react';
+import { classNamesFunction, styled, TextField } from '@fluentui/react';
+import { useExtendedTheme } from '../../../../Models/Hooks/useExtendedTheme';
+import { getDebugLogger } from '../../../../Models/Services/Utils';
+import {
+    IStoreFormModalProps,
+    IStoreFormModalStyleProps,
+    IStoreFormModalStyles
+} from './StoreFormModal.types';
+import { getStyles } from './StoreFormModal.styles';
+import CardboardModal from '../../../../Components/CardboardModal/CardboardModal';
+import { ITargetDatabaseConnection } from '../../Contexts/AppDataContext/AppDataContext.types';
+import { useTranslation } from 'react-i18next';
+import produce from 'immer';
+
+const debugLogging = false;
+const logDebugConsole = getDebugLogger('StoreFormModal', debugLogging);
+
+const getClassNames = classNamesFunction<
+    IStoreFormModalStyleProps,
+    IStoreFormModalStyles
+>();
+
+const EMPTY_DATABASE_CONNECTION: ITargetDatabaseConnection = {
+    clusterUrl: '',
+    databaseName: ''
+};
+
+const StoreFormModal: React.FC<IStoreFormModalProps> = (props) => {
+    const { isOpen, onConfirm, onDismiss, styles } = props;
+
+    // contexts
+
+    // state
+    const [
+        databaseConnection,
+        setDatabaseConnection
+    ] = useState<ITargetDatabaseConnection>(EMPTY_DATABASE_CONNECTION);
+
+    // hooks
+    const { t } = useTranslation();
+
+    // data
+    const isFormValid = useMemo(() => {
```

Comment:
this isn't particularly expensive, I don't think we need the useMemo

Reply:
that's fair. Will take a look.

Reply:
Done